### ISSUE TITLE
Generic notification on track change w/ KDE Plasma

### ISSUE THREAD

**Issue by Laurence-NZ** [Taiko2k/Tauon#1047]
Installed with Flatpak 
v7.6.7
on Kubuntu 23.04
w/ KDE Plasma 5.27.4 

When the application does not have focus and the track changes, it displays a generic notification under 'Plasma Workspace'. Is this supposed to happen? I would assume that it would have its own application slot so I could customise it's behaviour in KDE settings. I also would assume that the album art should be shown instead of a generic KDE logo. 

Currently I personally have Plasma Workspace notifications set to make a sound when they pop... obviously this is not the setting I want applied for music app notifications - so I have disable this effect globally just for Tauon. These notifications stack up and pollute the notification drawer unless I disable history for all Plasma Workspace notifications as well.

Love the application and greatly appreciate the amazing work you have done already!

_Example Notification_
![image](https://github.com/Taiko2k/TauonMusicBox/assets/48636474/eece5832-ce20-4278-9e5e-a204e57f5de6)

_KDE Notification Settings_
![image](https://github.com/Taiko2k/TauonMusicBox/assets/48636474/99c91f61-804a-43ff-a934-4f63e3c80023)

**Comment by Taiko2k** [comment]
You can disable those notifications in Tauon by going to MENU > Settings > Window > and uncheck "Emit Track Change Notifications".

The album art is supposed to show. Works when I test under GNOME. And not sure why it doesn't get its own application slot.

**Comment by Laurence-NZ** [comment]
Thanks for the tip. I'll disable them until they behave properly in KDE again.

**Comment by C0rn3j** [comment]
@Laurence-NZ I presume this is not a problem anymore on Plasma 6.x?

**Comment by Laurence-NZ** [comment]
> @Laurence-NZ I presume this is not a problem anymore on Plasma 6.x?

Haven't upgraded yet, but will report back!

**Comment by Laurence-NZ** [comment]
Sadly the issue is still present on Plasma 6.1.5

![image](https://github.com/user-attachments/assets/aa2ce157-3c10-4989-b816-34521ccfef10)

**Comment by C0rn3j** [comment]
What distro/version and how are you installing Tauon?

![image](https://github.com/user-attachments/assets/cbf3cd43-b0a5-407f-be48-6415b8871b39)
![image](https://github.com/user-attachments/assets/fc43c59a-0867-472e-8c7f-aed402b6cf29)

![image](https://github.com/user-attachments/assets/e1ffed41-657c-4241-be20-ed8c9ec77b6a)

**Comment by Laurence-NZ** [comment]
![image](https://github.com/user-attachments/assets/1a5455c9-5688-4335-9003-5f00e3a1644b)

Flatpak

**Comment by C0rn3j** [comment]
So the album picture does not show for me, which may be a portal limitation(?), but the icon and app name(albeit different... are we setting TMB normally vs just plain Tauon on Flatpak?) is picked up.

![image](https://github.com/user-attachments/assets/ee0c3b73-2ff2-437f-8d00-3fc0a2530d2f)

![image](https://github.com/user-attachments/assets/7f16418f-2959-4094-8f68-1ed4afdf2da9)

Can't tell if your system which is a couple months behind on mine with SW versions is the issue or something else is at play, but it does work for me even on Flatpak - and I made sure I uninstalled my system version to not accidentally provdie something there.

**Comment by Laurence-NZ** [comment]
Will try on a new user...

Your notifications match mine:
![image](https://github.com/user-attachments/assets/a166bd82-4122-4ad5-a6d0-ce796b2befae)

The original reason I logged this issue was that these are treated as 'system' notifications. This wasn't always the case. Do you mind clicking the settings cog and see if they are 'system' notifications for your as well? Ideally they should be shown as their own separate app notifications.

![image](https://github.com/user-attachments/assets/43104782-e04e-451c-b98e-c87e7004584e)

The reason this is an issue is because I can't mute the notifications without muting all system notifications. It would also be preferable to me if these notifications were ephemeral and didn't fill up the notification drawer.

**Comment by Laurence-NZ** [comment]
Confirmed same behaviour/problem on a new system user

**Comment by C0rn3j** [comment]
> Ideally they should be shown as their own separate app notifications.

Do you have a Flatpak application that works this way?

Something tells me that portal notifications just work this way without separate entries, but maybe I am wrong and Tauon does something wrong.

**Comment by Laurence-NZ** [comment]
I don't have an example on hand. This may indeed just be a limitation of Flatpak apps, I'm not sure.

I still think having the system notification sound pop every track change is not ideal. And neither is muting all system notification sounds to workaround this issue. Thank you for your help, maybe this should be a feature request instead of a bug report. 

I'm considering switching to arch soon - are they logged as system notifications with the AUR version?

**Comment by C0rn3j** [comment]
> I'm considering switching to arch soon - are they logged as system notifications with the AUR version?

Yup, that's a normal unsandboxed version, and it has a separate Application definition.

Though it would be best to solve this in some way, for example even if this turns out to be the default, expected behavior, you might still want to get Plasma to implement name-based filtering to be able to do what you want.
